apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/ServerInfoController.java

```diff
@@ -0,0 +1,219 @@
+/*
+ *  Copyright © 2018-2020 Apollo Foundation
+ */
+package com.apollocurrency.aplwallet.apl.core.rest.endpoint;
+
+import javax.annotation.security.PermitAll;
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+
+import java.util.Map;
+import java.util.Objects;
+
+import com.apollocurrency.aplwallet.api.dto.info.AccountsCountDto;
+import com.apollocurrency.aplwallet.api.dto.info.BlockchainConstantsDto;
+import com.apollocurrency.aplwallet.api.dto.info.BlockchainStateDto;
+import com.apollocurrency.aplwallet.api.dto.info.BlockchainStatusDto;
+import com.apollocurrency.aplwallet.api.dto.info.TimeDto;
+import com.apollocurrency.aplwallet.api.dto.info.TotalSupplyDto;
+import com.apollocurrency.aplwallet.apl.core.rest.RestParameters;
+import com.apollocurrency.aplwallet.apl.core.rest.service.ServerInfoService;
+import com.apollocurrency.aplwallet.apl.core.rest.utils.ResponseBuilder;
+import com.apollocurrency.aplwallet.apl.util.Constants;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Path("/server")
+public class ServerInfoController {
+    private ServerInfoService serverInfoService;
+    @Context SecurityContext context;
+
+    @Inject
+    public ServerInfoController(ServerInfoService serverInfoService) {
+        this.serverInfoService = Objects.requireNonNull(serverInfoService,"serverInfoService is NULL");
+    }
+
+    public ServerInfoController() {
+    }
+
+    @Path("/info/count")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(
+        summary = "Returns statistics Account information",
+        description = "Returns statistics information about specified count of account",
+        tags = {"info"},
+        responses = {
+            @ApiResponse(responseCode = "200", description = "Successful execution",
+                content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = AccountsCountDto.class)))
+        }
+    )
+    @PermitAll
+    public Response counts(
+        @Parameter(name = "numberOfAccounts", description = "number Of returned Accounts", allowEmptyValue = true)
+            @QueryParam("numberOfAccounts") String numberOfAccountsStr
+    ) {
+        log.debug("Started counts : \t'numberOfAccounts' = {}", numberOfAccountsStr);
+        ResponseBuilder response = ResponseBuilder.startTiming();
+        int numberOfAccounts = RestParameters.parseInt(numberOfAccountsStr, "numberOfAccounts",
```

Comment:
Looks like better to use bean validation for such purposes

Reply:
Updated